src/transport/Session.h

```diff
@@ -225,6 +228,19 @@ class Session
 
     bool IsUnauthenticatedSession() const { return GetSessionType() == SessionType::kUnauthenticated; }
 
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    ActiveConnectionState * GetTCPConnection() const { return mTCPConnection; }
+    // This API is used to associate the connection with the session when the
+    // latter is about to be marked active. It is also used to reset the
+    // connection to a nullptr when the connection is lost and the session
+    // is marked as Defunct.
+    // The lifetime of this member connection pointer is, essentially the same
+    // as that of the underlying connection with the peer.
+    // It would remain as a nullptr for all sessions that are not set up over
+    // a TCP connection.
```

Comment:
This part, and probably the lifetime part, belong with the getter, no?

Reply:
Consolidated the Getter and Setter together.